extension/agenthealth/handler/stats/agent/flag.go

```diff
@@ -0,0 +1,188 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package agent
+
+import (
+	"encoding"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+var (
+	errUnsupportedFlag = errors.New("unsupported usage flag")
+)
+
+const (
+	FlagIMDSFallbackSuccess Flag = iota
+	FlagSharedConfigFallback
+	FlagAppSignal
+	FlagEnhancedContainerInsights
+	FlagRunningInContainer
+	FlagMode
+	FlagRegionType
+
+	flagIMDSFallbackSuccessStr       = "imds_fallback_success"
+	flagSharedConfigFallbackStr      = "shared_config_fallback"
+	flagAppSignalsStr                = "app_signals"
+	flagEnhancedContainerInsightsStr = "enhanced_container_insights"
+	flagRunningInContainerStr        = "running_in_container"
+	flagModeStr                      = "mode"
+	flagRegionTypeStr                = "region_type"
```

Comment:
qq: what is region_type? Is it the region code?

Reply:
I think it's how the agent got the region. https://github.com/aws/amazon-cloudwatch-agent/blob/main/translator/config/regiontype.go